lib/main.rb

```diff
@@ -0,0 +1,11 @@
+def substrings(text, dictionary)
+  matches = {}
+
+  dictionary.each do |key|
+    matches_count = text.scan(/(#{key})/ix).size
+
+    matches[key] = (matches[key] || 0) + matches_count if matches_count.positive?
+  end
+
+  matches
+end
```

Comment:
Nice solution! I just suggest one small change that is useful to know: in ruby, there is a way to instantiate a hash, providing a default value for any new key you add to the hash. Instead of doing `matches = {}` you can use `matches = Hash.new(0)` and 0 will be the default value for any new entry. That way you save that `(matches[key] || 0)` operation later on

```suggestion
def substrings(text, dictionary)
  matches = Hash.new(0)

  dictionary.each do |key|
    matches_count = text.scan(/(#{key})/ix).size

    matches[key] += matches_count if matches_count.positive?
  end

  matches
end
```

Reply:
Interesting, because if I use `matches={}` instead of `matches = Hash.new(0)`, `matches[key] += matches_count` throw an error. Why??

Reply:
this is why I used:
`matches[key] = (matches[key] || 0) + matches_count`

Reply:
Done!

Reply:
> Interesting, because if I use matches={} instead of matches = Hash.new(0), matches[key] += matches_count throw an error. Why??

When you instantiate a hash like this `matches = {}`, every new key that you try to access to the hash will have `nil` as value
<img width="311" alt="image" src="https://user-images.githubusercontent.com/27536621/188993762-8c264c4b-f5d5-407c-a3b0-5ef8280f8545.png">

If you do `matches = Hash.new(0)` the default value will be 0, not nil

Reply:
More info on different ways to create a hash here: https://ruby-doc.org/core-3.1.2/Hash.html#class-Hash-label-Creating+a+Hash